src/movies.js

```diff
@@ -1,28 +1,70 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+
+const movies = require("./data");
+
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+  const directors = moviesArray.map(function (element) {
+    return element.director;
+  })
+  return directors;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  const spielbiergMovies = moviesArray.filter(function (element) {
+    return element.director === 'Steven Spielberg';
+  })
+
+  const dramaMovies = spielbiergMovies.filter(function (element) {
+    return element.genre.includes("Drama");
+  })
+  return dramaMovies.length;
+}
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  if (moviesArray.length === 0) {
+    return 0;
+  }
+  const sum = moviesArray.filter((current) => current.score >= 0).reduce(function (accumulator, current) {
+    return accumulator + current.score;
+  }, 0)
+  const average = (Math.round((sum / moviesArray.length) * 100) / 100);
+  return average;
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+  if (moviesArray.length === 0) {
+    return 0;
+  }
```

Comment:
For shorter syntax, if you return only one line (literally) , you can use below syntax: 

```
  if (moviesArray.length === 0) return 0;
```